core/src/main/java/com/datastrato/gravitino/storage/relational/utils/SessionUtils.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.storage.relational.utils;
+
+import com.datastrato.gravitino.storage.relational.session.SqlSessions;
+import java.util.Arrays;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import org.apache.ibatis.session.SqlSession;
+
+/**
+ * This class provides utility methods to perform database operations with MyBatis mappers in the
+ * SqlSession.
+ */
+public class SessionUtils {
+  private SessionUtils() {}
+
+  /**
+   * This method is used to perform a database operation with a commit. If the operation fails, the
+   * transaction will roll back.
+   *
+   * @param mapperClazz mapper class to be used for the operation
+   * @param consumer the operation to be performed with the mapper
+   * @param <T> the type of the mapper
+   */
+  public static <T> void doWithCommit(Class<T> mapperClazz, Consumer<T> consumer) {
+    try (SqlSession session = SqlSessions.getSqlSession()) {
+      try {
+        T mapper = SqlSessions.getMapper(mapperClazz);
+        consumer.accept(mapper);
+        SqlSessions.commitAndCloseSqlSession();
+      } catch (Throwable t) {
+        SqlSessions.rollbackAndCloseSqlSession();
+        throw new RuntimeException(t);
+      }
+    }
```

Comment:
Why don't we implement it like this? I noticed that the function getMapper first obtains the session and then proceeds with the getMapper logic. In line 38, we have already obtained the session, and the other functions are the same.
```
  public static <T> void doWithCommit(Class<T> mapperClazz, Consumer<T> consumer) {
    try (SqlSession session = SqlSessions.getSqlSession()) {
      try {
        T mapper = session.getMapper(mapperClazz);
        consumer.accept(mapper);
        session.commitAndCloseSqlSession();
      } catch (Throwable t) {
        session.rollbackAndCloseSqlSession();
        throw new RuntimeException(t);
      }
    }
```
Is using a session pool to manage the sessions better than using ThreadLocal?

Reply:
Yeah, I think the logic of `getMapper` can be simplified. But we don’t need to pool SqlSession here, `MyBatis` will have its own processing logic in `SqlSessionFactory`.

Reply:
The `getMapper()` method still needs to be retained. When multiple operations are required to ensure transactions at the same time, we pass the session through ThreadLocal.

Reply:
We are only using one session in the doWithCommit function and passing it to the getMapper, commitAndCloseSqlSession, and rollbackAndCloseSqlSession functions. Why do you obtain the session from the ThreadLocal in these three function calls?

Reply:
If I understand you correctly, please take a look at the relevant logic of `doMultipleWithCommit`. When performing multiple database operations here, in order to shield the upper layer from the perception of `SqlSession`, we call `doWithoutCommit` to obtain the same Session from ThreadLocal. The implementations in other places are similar. I do not want to transfer Sessions between functions and try to keep their respective implementations independent.

Reply:
This may just be a personal preference.